src/audit-log/src/lib.rs

```diff
@@ -233,35 +239,20 @@ impl EventV1 {
 // failing. Instead of changing data structures, add new variants.
 #[test]
 fn test_audit_log() -> Result<(), anyhow::Error> {
-    let cases: Vec<(VersionedEvent, &'static str)> = vec![
-        (
-            VersionedEvent::V1(EventV1::new(
-                1,
-                EventType::Create,
-                ObjectType::View,
-                EventDetails::NameV1(NameV1 {
-                    name: "name".into(),
-                }),
-                Some("user".into()),
-                1,
-            )),
-            r#"{"V1":{"id":1,"event_type":"create","object_type":"view","event_details":{"NameV1":{"name":"name"}},"user":"user","occurred_at":1}}"#,
-        ),
-        (
-            VersionedEvent::V1(EventV1::new(
-                2,
-                EventType::Drop,
-                ObjectType::ClusterReplica,
-                EventDetails::IdNameV1(IdNameV1 {
-                    id: "u1".to_string(),
-                    name: "name".into(),
-                }),
-                None,
-                2,
-            )),
-            r#"{"V1":{"id":2,"event_type":"drop","object_type":"cluster-replica","event_details":{"IdNameV1":{"id":"u1","name":"name"}},"user":null,"occurred_at":2}}"#,
-        ),
-    ];
+    let cases: Vec<(VersionedEvent, &'static str)> = vec![(
+        VersionedEvent::V1(EventV1::new(
+            2,
+            EventType::Drop,
+            ObjectType::ClusterReplica,
+            EventDetails::IdNameV1(IdNameV1 {
+                id: "u1".to_string(),
+                name: "name".into(),
+            }),
+            None,
+            2,
+        )),
+        r#"{"V1":{"id":2,"event_type":"drop","object_type":"cluster-replica","details":{"IdNameV1":{"id":"u1","name":"name"}},"user":null,"occurred_at":2}}"#,
```

Comment:
Test failure complaining here, got:
`"details":{"id":"u1","name":"name"}`
expected:
`"details":{"IdNameV1":{"id":"u1","name":"name"}}`
Seems like the `flatten` flattened the outer struct?